purchase_split_wizard_deposit/wizards/purchase_split_wizard.py

```diff
@@ -0,0 +1,128 @@
+                            "product_id": line.product_id,
+                            "name": line.name,
+                            "purchase_line_origin_id": line.id,
+                            "display_type": line.display_type,
+                        },
+                    )
+                    for line in splitteable_advance_lines
+                ],
+            }
+        )
+        return res
+
+    def _get_new_purchase_vals(self):
+        res = super()._get_new_purchase_vals()
+        if self.split_advances:
+            res["order_line"] += [
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": line.product_id.id,
+                        "name": line.name,
+                        "product_qty": 0,
+                        "qty_splitted": line.amount,
+                        "price_unit": line.amount,
+                        "splitted_purchase_line_parent_id": (
+                            line.purchase_line_origin_id.id
+                        ),
+                        "date_planned": line.purchase_line_origin_id.date_planned,
+                        "display_type": line.display_type,
+                        "is_deposit": True,
+                    },
+                )
+                for line in self.advance_line_ids
+                if line.amount or line.display_type
+            ]
+        return res
+
+
+class PurchaseAdvanceLineSplitWizard(models.TransientModel):
+    _name = "purchase.advance.line.split.wizard"
+    _description = "Wizard aux model to split purchase lines"
+
+    wizard_id = fields.Many2one(
+        string="Wizard",
+        comodel_name="purchase.split.wizard",
+    )
+    purchase_line_origin_id = fields.Many2one(
+        string="Purchase Line", comodel_name="purchase.order.line", readonly=True
+    )
+    product_id = fields.Many2one(
+        string="Product",
+        comodel_name="product.product",
+    )
+    name = fields.Text()
+    currency_id = fields.Many2one(
+        related="purchase_line_origin_id.order_id.currency_id",
+        string="Currency",
+        readonly=True,
+    )
+    amount = fields.Monetary()
+    max_amount = fields.Monetary(compute="_compute_max_amount", readonly=True)
+    display_type = fields.Selection(
+        selection=[("line_section", "Section"), ("line_note", "Note")],
+        default=False,
+    )
+
+    @api.depends("purchase_line_origin_id")
+    def _compute_max_amount(self):
+        for rec in self:
+            rec.max_amount = rec.purchase_line_origin_id.qty_splittable
+
+    @api.constrains("amount", "max_amount")
+    def _check_max_amount(self):
+        for rec in self:
+            if not rec.display_type and rec.amount > rec.max_amount:
+                raise exceptions.ValidationError(
+                    _(
+                        "The amount can not be higher than the max amount.\n"
+                        f"- line: {rec.name}\n"
```

Comment:
I think that using f functions prevent strings from being translatable. Note that this string in not in the translations file.

Reply:
Solved!